app/src/main/java/edu/uc/it3048/mypokedex/ProfileScreenActivity.kt

```diff
@@ -156,39 +146,47 @@ class ProfileScreenActivity : AppCompatActivity() {
             .document()
             .set(location)
 
-            // TODO add log statement?
+        // TODO add log statement?
     }
 
-    @SuppressLint("SimpleDateFormat")
-    private fun createImageFile() : File {
-        val timeStamp = SimpleDateFormat("yyyyMMdd_HHmmss".format(Date()))
-        val storageDir : File? = this.getExternalFilesDir(Environment.DIRECTORY_PICTURES)
-        return File.createTempFile("MyPokedex${timeStamp}", ".jpg", storageDir).apply {
-            currentPhotoPath = absolutePath
+    // Method to view saved pokemon sighting locations
+    private fun savedLocationsFolder(){
+        val folderButton = findViewById<ImageButton>(R.id.imgBtnFolder)
+        folderButton.setOnClickListener {
+            val savedLocationsIntent = Intent(this, SavedSightingsActivity::class.java)
+            startActivity(savedLocationsIntent)
         }
     }
+    //endregion
+
+    //region Image Functions
 
     private fun prepTakePhoto() {
-        if (ContextCompat.checkSelfPermission(this, android.Manifest.permission.CAMERA) == PackageManager.PERMISSION_GRANTED) {
-            takePhoto()
-        } else {
-            val permissionRequest = arrayOf(android.Manifest.permission.CAMERA)
-            requestPermissions(permissionRequest, CAMERA_REQUEST_CODE)
+        try {
+            if (ContextCompat.checkSelfPermission(this, android.Manifest.permission.CAMERA) == PackageManager.PERMISSION_GRANTED) {
+                takePhoto()
+            } else {
+                val permissionRequest = arrayOf(android.Manifest.permission.CAMERA)
+                requestPermissions(permissionRequest, CAMERA_REQUEST_CODE)
+            }
+        }
+        catch(e: IOException) {
+            error(e.message!!)
```

Comment:
Added a try/catch block for error handling. This can be implemented across other functions if that would be suitable for the intentions of the group. Also, the contents of the catch block can be used differently, such as logging errors.

Reply:
If you add a try-catch block, do something with the catch:
- Try to fix the issue.
- Definitely log the issue.
- Figure out where to go next.  Should you continue with the next line of logic, or is that impossible to do?  If impossible, re-throw the exception.

While the exception block is here, it isn't doing any of the above items, so it's not fully complete.  It doesn't add value until fully complete.  As a matter of fact, it can make things worse, because it effectively ignores the error, and acts like nothing went wrong.